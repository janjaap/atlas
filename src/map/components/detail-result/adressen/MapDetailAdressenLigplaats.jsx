import React from 'react';
import PropTypes from 'prop-types';

import MapDetailResultItem from '../MapDetailResultItem';
import MapDetailResultWrapper from '../MapDetailResultWrapper';

const MapDetailAdressenLigplaats = ({
  panoUrl,
ligplaats,
onMaximize,
onPanoPreviewClick
}) => (
  <MapDetailResultWrapper
    onMaximize={onMaximize}
    onPanoPreviewClick={onPanoPreviewClick}
    panoUrl={panoUrl}
    subTitle={ligplaats.label}
    title="Ligplaats"
  >
    <ul className="map-detail-result__list">
      <MapDetailResultItem
        label="Status"
        value={ligplaats.status.description}
      />
    </ul>
  </MapDetailResultWrapper>
);

MapDetailAdressenLigplaats.propTypes = {
  ligplaats: PropTypes.shape({
    label: PropTypes.string,
    status: PropTypes.shape({
      description: PropTypes.string,
      code: PropTypes.string
    }).isRequired
  }).isRequired,
  panoUrl: PropTypes.string.isRequired,
  onMaximize: PropTypes.func.isRequired,
  onPanoPreviewClick: PropTypes.func.isRequired
};

export default MapDetailAdressenLigplaats;
