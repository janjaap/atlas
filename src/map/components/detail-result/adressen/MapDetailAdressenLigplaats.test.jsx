import React from 'react';
import { shallow } from 'enzyme';

import MapDetailAdressenLigplaats from './MapDetailAdressenLigplaats';

describe('MapDetailAdressenLigplaats', () => {
  it('should render everything', () => {
    const ligplaats = {
      label: 'Ligplaats label',
      status: { description: 'description', code: '18' }
    };
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <MapDetailAdressenLigplaats
        panoUrl="panoUrl"
        onMaximize={clickHandler}
        onPanoPreviewClick={clickHandler}
        ligplaats={ligplaats}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
