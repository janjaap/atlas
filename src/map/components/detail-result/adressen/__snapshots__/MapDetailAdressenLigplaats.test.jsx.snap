// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapDetailAdressenLigplaats should render everything 1`] = `
<MapDetailResultWrapper
  onMaximize={[MockFunction]}
  onPanoPreviewClick={[MockFunction]}
  panoUrl="panoUrl"
  subTitle="Ligplaats label"
  title="Ligplaats"
>
  <ul
    className="map-detail-result__list"
  >
    <MapDetailResultItem
      label="Status"
      value="description"
    />
  </ul>
</MapDetailResultWrapper>
`;
